### ISSUE TITLE
Deadlock while grouping on CSV file

### ISSUE THREAD

**Issue by fbirlik** [duckdb/duckdb#8890]
### What happens?

Simple group by query on 40k line CSV file gets stuck indefinitely.

Works with 0.8.1, 0.7.1, fails with latest source version.

Also did git bisect, it looks like following commit is when the code starts to get stuck.

[55a16eb263b1f7449a65ac97ec06d4f904c822d4] use lock instead of atomic for combined threads

### To Reproduce

build latest code and run following sql query against attached file

[hebere.csv.gz](https://github.com/duckdb/duckdb/files/12582417/hebere.csv.gz)

```
select
    field1,
    field2,
    min(datum) as firstseen,
    max(datum) as lastseen
from read_csv('./hebere.csv.gz',
        delim='\t',
        header=0,
        columns={field1: varchar, field2: varchar, datum: varchar},
        compression='gzip',
        ignore_errors=0)
group by field1, field2
limit 10;
```

### OS:

ubuntu 22.04

### DuckDB Version:

v0.8.2-dev4305 9db510bd11

### DuckDB Client:

cli

### Full Name:

Firat Birlik

### Affiliation:

Airties

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by pdet** [comment]
Hi @fbirlik, thanks for testing the latest master, the CSV parser went through quite a big change, so it's nice to see people giving it a go :-)!

 It seems I already fixed this issue on my pr #8852 
Could you confirm that's the case? In any case, I've also added the test and data you sent to that PR.

**Comment by fbirlik** [comment]
@pdet you are indeed correct.

I confirm that https://github.com/duckdb/duckdb/pull/8852 fixes the issue!
